hrms/overrides/expense_claim.py

```diff
@@ -0,0 +1,350 @@
+                }
+            )
+            doc = user_permission_doc.insert()
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def next_state(doc_name):
+    expense_claim_doc = frappe.get_doc("Expense Claim", doc_name)
+    if expense_claim_doc.status == ExpenseClaimConstants.DRAFT:
+        expense_claim_doc.status = ExpenseClaimConstants.PENDING_APPROVAL
+        expense_claim_doc.save()
+    elif expense_claim_doc.status == ExpenseClaimConstants.PENDING_APPROVAL:
+        expense_claim_doc.status = ExpenseClaimConstants.PENDING_APPROVAL_BY_ADMIN
+        expense_claim_doc.save()
+        expense_claim_doc.submit()
+    elif expense_claim_doc.status == ExpenseClaimConstants.PENDING_APPROVAL_BY_ADMIN:
+        expense_claim_doc.status = ExpenseClaimConstants.PENDING_APPROVAL_BY_HR
+        expense_claim_doc.save()
+    elif expense_claim_doc.status == ExpenseClaimConstants.PENDING_APPROVAL_BY_HR:
+        if expense_claim_doc.grand_total == 0:
+            expense_claim_doc.status = ExpenseClaimConstants.PAID
+        else:
+            expense_claim_doc.status = ExpenseClaimConstants.PENDING_PAYMENT
+        expense_claim_doc.save()
+    else:
+        frappe.throw(_("Unhandelled State Transition"))
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def get_total_pending_amount(doc_employee_name):
+    expense_claims = frappe.db.get_list(
+        "Expense Claim",
+        filters={"status": "Pending Payment", "employee_name": doc_employee_name},
+        fields=["grand_total"],
+        pluck="grand_total",
+    )
+    total_amount = sum(expense_claims)
+    return total_amount
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def get_mode_of_payments():
+    return frappe.db.get_list("Mode of Payment", pluck="name")
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def create_payment_entry(doc_name, values):
+    if not user_has_role(frappe.session.user, RoleConstants.FINANCE_ROLE):
+        frappe.throw(_("Only Finance Team can make a Payment entry"))
+
+    doc = frappe.get_doc("Expense Claim", doc_name)
+
+    import json
+
+    if isinstance(values, str):
+        payment_values = frappe._dict(json.loads(values))
+
+    party_details = get_party_details(
+        doc.company, "Employee", doc.employee, frappe.utils.nowdate()
+    )
+
+    bank_cash_account = get_bank_cash_account(payment_values.mode_of_payment, doc.company).get(
+        "account"
+    )
+    bank_account_currency = get_account_details(bank_cash_account, frappe.utils.nowdate()).get(
+        "account_currency"
+    )
+
+    paid_to = doc.payable_account
+    paid_to_account_currency = get_account_details(paid_to, frappe.utils.nowdate()).get(
+        "account_currency"
+    )
+
+    expense_claims_list = frappe.db.get_list(
+        "Expense Claim",
+        filters={"status": "Pending Payment", "employee_name": doc.employee_name},
+        pluck="name",
+    )
```

Comment:
receive list of expense claims from front end. Front end should have received those when the dialog box is opened.

Validate sum of sanctioned amounts in list of expense claims is equal to total amount received in the api

Reply:
Done